sklearn/exceptions.py

```diff
@@ -154,3 +155,12 @@ class UndefinedMetricWarning(UserWarning):
     .. versionchanged:: 0.18
        Moved from sklearn.base.
     """
+
+
+class SklearnDeprecationWarning(DeprecationWarning):
```

Comment:
I would not derive from DeprecationWarning but use UserWarning like numpy, this way you don't need the `warnings.filterwarnings` hack in `sklearn/__init__.py`. See https://github.com/scikit-learn/scikit-learn/pull/7995#issuecomment-265418647 for more details.